src/components/molecules/VenueRoomItem/VenueRoomItem.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { useMemo } from "react";
+import { Button, Form, Modal } from "react-bootstrap";
+import { useForm } from "react-hook-form";
+import { useAsyncFn } from "react-use";
+
+import { DEFAULT_VENUE_LOGO } from "settings";
+
+import { createRoom, createVenue_v2 } from "api/admin";
+
+import { venueInsideUrl } from "utils/url";
+import { buildEmptyVenue } from "utils/venue";
+
+import { RoomTemplate, VenueRoomTemplate } from "types/rooms";
+
+import { useUser } from "hooks/useUser";
+import { useVenueId } from "hooks/useVenueId";
+import { useShowHide } from "hooks/useShowHide";
+
+import {
+  venueRoomSchema,
+  roomSchema,
+} from "pages/Admin/Details/ValidationSchema";
+
+import { InputField } from "components/atoms/InputField";
+
+import "./VenueRoomItem.scss";
+
+export interface VenueRoomItemProps {
+  icon: string;
+  text: string;
+  template?: VenueRoomTemplate;
+}
+
+export const VenueRoomItem: React.FC<VenueRoomItemProps> = ({
+  icon,
+  text,
+  template,
+}) => {
+  const {
+    isShown: isModalVisible,
+    show: showModal,
+    hide: hideModal,
+  } = useShowHide();
+
+  const { user } = useUser();
+
+  const venueId = useVenueId();
+
+  const isVenuePortal = template !== RoomTemplate.external;
+
+  const { register, getValues, handleSubmit, errors } = useForm({
+    validationSchema: isVenuePortal ? venueRoomSchema : roomSchema,
+    defaultValues: {
+      roomTitle: "",
+      roomUrl: "",
+      venueName: "",
+      template: template,
+    },
+  });
+
+  const [{ loading }, addRoom] = useAsyncFn(async () => {
+    if (!user || !venueId || !template) return;
+
+    const roomValues = getValues();
+
+    const roomUrl = isVenuePortal
+      ? window.origin + venueInsideUrl(roomValues.venueName)
+      : roomValues.roomUrl;
+
+    const roomData = {
+      title: roomValues.roomTitle,
+      isEnabled: true,
+      image_url: DEFAULT_VENUE_LOGO,
+      url: roomUrl,
+      template,
+    };
+
+    // TS doesn't work properly with const statements and won't 'know' that this is already checked.
+    // That's why this is inline instead of isVenuePortal
+    if (template !== RoomTemplate.external) {
+      const venueData = buildEmptyVenue(roomValues.venueName, template);
+
+      await createVenue_v2(venueData, user);
+    }
+
+    await createRoom(roomData, venueId, user).then(() => hideModal());
+  }, [getValues, hideModal, isVenuePortal, template, user, venueId]);
+
+  const roomIconStyles = useMemo(() => ({ backgroundImage: `url(${icon})` }), [
+    icon,
+  ]);
```

Comment:
I think you can use <image /> with src prop to avoid this kind of code

Reply:
Fixed in: https://github.com/sparkletown/sparkle/pull/1583/commits/f8fa0c67c34ac626919737b4bbd112b4bd715553